libraries/DataFlash/LogStructure.h

```diff
@@ -873,6 +873,18 @@ struct PACKED log_Proximity {
     float closest_dist;
 };
 
+struct PACKED log_SRTL {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t active;
+    uint16_t num_points;
```

Comment:
At first I had `uint`s everywhere. But then I decided to change `_simplify_stack_last_index`, `_last_index` and `_prunable_loops_last_index`, in order to allow them to store a value of -1 (meaning that there was is not a single thing in the arrays they represent). But then the compiler got angry at me for lots of comparisons between `uint`s and `int`s, so I changed pretty much everything.

So maybe the best fix is to change everything to a `uint` and find some better way to represent that an array is empty.